src/app/chat/views/messages.nim

```diff
@@ -496,3 +479,46 @@ QtObject:
     {
       ChatViewRoles.MessageList.int:"messages"
     }.toTable
+
+  proc getSearchResultMessageModel*(self: MessageView): QVariant {.slot.} = 
+    newQVariant(self.searchResultMessageModel)
+
+  # we just need to expose model to qml, there is no need for exposing notifying signal
+  QtProperty[QVariant] searchResultMessageModel:
+    read = getSearchResultMessageModel
+
+  proc onSearchMessages*(self: MessageView, response: string) {.slot.} =
+    let responseObj = response.parseJson
+    if (responseObj.kind != JObject):
+      error "search messages response is not an json object"
+      return
+
+    let chatId = if(responseObj.contains("chatId")): responseObj{"chatId"}.getStr else : ""
+    if (chatId.len == 0):
+      error "search messages response either doesn't contain chatId or it is empty"
+      return
+
+    let messagesObj = if(responseObj.contains("messages")): responseObj{"messages"} else: newJObject()
+    if (messagesObj.kind != JObject):
+      error "search messages response either doesn't contain messages object or it is empty"
+      return
+
+    let (cursor, messages) = libstatus_chat.parseChatMessagesResponse(messagesObj)
```

Comment:
I know using `libstatus` directly in a view is not recommended. Maybe we could expose `parseChatMessagesResponse` in `status/chat.nim` instead

Reply:
Yes, I fully agree, there are multiple places where we call libstatus procedures directly from the view. I will update all such calls in messages.nim, won't go through the whole app as this is not refactoring (but every time we are fixing/developing something is an opportunity for some kind of mini refactoring).

Reply:
Good idea!